Add attachments for OpenStreetMapParser with default license (#4710)
GeotrekCE/Geotrek-admin: docs/import-data/import-touristic-data-systems.rst

*means*: return objects that either have both ``boundary=administrative`` AND ``admin_level=4``, OR have ``highway=bus_stop``.

> 

Reviewer: Now that the doc for OSM parsers is gettng a bit long, it might be good to add a bit of structure by adding some titles.
I would add something like "Handling translated fields" here, and "Query configuration" either on line 278 or 283.

> Finally all the objects parsed by the OpenStreetMap parsers will be those contained in the ``settings.SPATIAL_EXTENT`` bounding box.

Reviewer: Either move this into the "Query configuration" section or make a separate section for this as well.

You can change the bounding box by overriding ``get_bbox_str()``.

> .. note::

Reviewer: If sections are created for the topics above, I think this should also be made into its own section, otherwise it could appear as being part of the above section.

   ``OpenStreetMapParser`` automatically attaches files from ``wikimedia_commons`` and ``image`` tags found in the data.
   A ``CC BY-SA 4.0`` license is assigned to each imported file, as specified by the OpenStreetMap license.

   For more information on how attachments work, consult :ref:`this section <import-attachments>`.

.. note::

>     When using the ``OpenStreetMapParser`` with the ``OpenStreetMap`` provider,
>     the eid field of the parsed objects in the detail sheet provide a link
>     to the corresponding object on the OpenStreetMap website.

Reviewer: ```suggestion
    When using ``OpenStreetMapParser`` with the ``OpenStreetMap`` provider,
    the ``eid`` field of the parsed objects in the detail sheet provide a link
    to the corresponding object on the OpenStreetMap website.
```

If you need to cancel the aggregation of portals, remove param ``m2m_aggregate_fields``.

.. _import-attachments:

> Import attachments

Reviewer: I think this section should be added between sections "Import from OSM" and "Multiple imports". Right now it's between "Multiple imports" and "Importing from multiple sources with deletion", which are kind of related.

The mixin is located in ``geotrek/common/parsers.py`` and must be inherited in your parser:

.. code-block:: python

   class ExampleParser(AttachmentParserMixin, Parser):

       # Parser configuration …

.. warning::

   Use ``AttachmentParserMixin`` **only in base parsers**.
   Custom parsers should focus on configuration.
   Factor attachment logic into shared base classes to keep custom parsers clean and maintainable.

Attributes
----------

The following attributes can be customized:

* ``download_attachments`` (default: ``True``):
  Whether to download and store attachments via Paperclip. If set to ``False``, attachments are only linked.
  Requires ``PAPERCLIP_ENABLE_LINK = True`` in Django settings.

* ``base_url`` (default: ``""``):
  Base URL prepended to each relative attachment path returned by ``filter_attachments``.

* ``delete_attachments`` (default: ``True``):
  After the new attachments have been processed, **every existing
  attachment that is *not* present in the current feed (or whose file has
  been replaced)** is permanently removed.

* ``filetype_name`` (default: ``"Photographie"``):
  Label of the ``FileType`` model assigned to all imported files.

>   If it does not exist in the database, the parser will raise a warning:

Reviewer: ```suggestion
  If it does not exist in the database, the import will fail with a warning:
```

  ::

     FileType '<name>' does not exist in Geotrek-Admin. Please add it

* ``non_fields`` (default: ``{"attachments": _("Attachments")}``):

>   Maps the internal ``attachments`` field to the field name containing attachment data in the external source.

Reviewer: ```suggestion
  Maps the internal ``attachments`` field to the field name(s) containing attachments data in the external source.
```

* ``default_license_label`` (default: ``None``):
  If specified, this license will be assigned to all imported attachments.
  If the license does not exist, it will be created automatically.

Filtering attachments
---------------------

> ``filter_attachments`` method format the external source data to match with the internal format.

Reviewer: ```suggestion
The ``filter_attachments`` method formats the external source data to match with the internal format.
```

Signature:

.. code-block:: python

   def filter_attachments(self, src, val) -> list[[str, str, str, str]]:
       ...
       return [[url, legend, author, title], ...]

Default implementation:

.. code-block:: python

   def filter_attachments(self, src, val):
       if not val:
           return []
       return [
           (subval.strip(), "", "", "")  # url, legend, author, title
           for subval in val.split(self.separator)
           if subval.strip()

>        ]

Reviewer: Should this be included?

> If the attachment data has a different structure ``filter_attachments`` should be override.

Reviewer: ```suggestion
If the attachment data has a different structure, ``filter_attachments`` should be overridden.
```
